File: Astronomy in OBJC/Astronomy in OBJC/Networking/NSURL+UsingHTTPS.m
//
//  NSURL+UsingHTTPS.m
//  Astronomy-in-OBJC
//
//  Created by Sal B Amer on 6/4/20.
//  Copyright © 2020 Sal B Amer. All rights reserved.
//

#import "NSURL+UsingHTTPS.h"

@implementation NSURL (UsingHTTPS)

- (NSURL *) usingHTTPS {
  NSURLComponents *components = [NSURLComponents componentsWithURL:self resolvingAgainstBaseURL:YES];
  components.scheme = @"https";
  return components.URL;
}

@end

File: LambdaSDK/NSURL+UsingHTTPS.swift
//
//  NSURL+UsingHTTPS.swift
//  Astronomy_ObjectiveC
//
//  Created by Kenny on 6/16/20.
//  Copyright © 2020 Hazy Studios. All rights reserved.
//

import Foundation

@objc extension NSURL {
    var usingHTTPS: NSURL? {
        guard var components = URLComponents(url: self as URL, resolvingAgainstBaseURL: true) else { return nil }
        components.scheme = "https"
        return components.url! as NSURL
    }
}
